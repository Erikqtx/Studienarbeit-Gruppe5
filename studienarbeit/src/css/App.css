@import url('https://fonts.googleapis.com/css2?family=Source+Sans+Pro&display=swap');

.App {
  background-color: #282c34;
  height: 100vh;
  display: grid;
  grid-template-columns: 3fr 2fr;
  font-family: 'Source Sans Pro', sans-serif;
}

.box {
  margin: 5px;
  background-color: aqua;
}

.header {
  grid-column: 1 / span 2;
  color: white;
  margin-top: 25px;
  text-align: center;
  width: 100%;
}

.text {
  grid-column: 1 / span 2;
  text-align: center;
  color: white;
}

.info {
  text-align: center;
  color: white;
  margin-left: 15px;
  border-left: 1px solid white;
}

@media only screen and (max-width: 768px) {
  .map {
    grid-column: 1 / span 2;
  }

  .info {
    grid-column: 1 / span 2;
    margin: 0;
    border-left: none;
    border-top: 1px solid white;
  }
}

