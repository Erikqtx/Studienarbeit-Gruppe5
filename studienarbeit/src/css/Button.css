/*Country-State Slider Button*/

.button-box1{
    width: 20%;
    margin-left: 10px;
    margin-top: 15px;
    position: relative;
    box-shadow: 9px 2px 16px -8px rgba(0,0,0,0.5);
    border-radius: 5px;
    background: rgba(195, 216, 221, 0.301);
    transform: translate(15%, 10%);
    overflow: hidden;
    float: left;
    
  }
  
  .button-box1 label {
    padding: 3px 0 3px 0;
    width: 50%;
    float: left;
    color:white;
    cursor: pointer;
    position: relative;
    font-size: 13px;
    text-align: center;
    font-family: 'Source Sans Pro', sans-serif;
  }
  
  
  .slider{
    position:absolute;
    left: 0%; 
    top:0;
    width:50%; 
    height: 100%;
    animation-timing-function: ease-in-out;
    animation-duration:.5s;
    transition: 0.1s left .05s ease-in-out;
    z-index: -1;
  }
 
  #country + label:before{
    content: "";
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 100%;
    z-index: -1;
    background-color: #fff;
    transition: all 0.3s ease;
  }
  
  #country:checked + label:before{
    left: 0;
  }
  
  #country:checked + label,
  #state:checked + label{
    color:black;
  }

/*Year 2015-2016 Slider Button*/

.button-box2{
  width: 20%;
  margin-right: 10%;
  margin-top: 15px;
  position: relative;
  box-shadow: 9px 2px 16px -8px rgba(0,0,0,0.5);
  border-radius: 5px;
  background: rgba(195, 216, 221, 0.301);
  transform: translate(15%, 10%);
  overflow: hidden; 
  float: right;

  
}

.button-box2 label {
    padding: 3px 0 3px 0;
    width: 50%;
    float: left;
    color:white;
    cursor: pointer;
    position: relative;
    font-size: 13px;
    text-align: center;
    font-family: 'Source Sans Pro', sans-serif;
  }

  #year1 + label:before{
    content: "";
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 100%;
    z-index: -1;
    background-color: #fff;
    transition: all 0.3s ease;
  }
  
  #year1:checked + label:before{
    left: 0;
  }
  
  #year1:checked + label,
  #year2:checked + label{
    color:black;
  }
  

  @media only screen and (max-width: 600px) {
    
    .button-box1,
    .button-box2{
      width: 30%;
    }
    
    .button-box1 label,
    .button-box2 label {
      font-size: 12px;
    }
  
  }