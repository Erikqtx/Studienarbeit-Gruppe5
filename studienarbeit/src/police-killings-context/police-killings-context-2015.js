import parse from "csv-parse/lib/sync";
import React, { useState, useEffect, createContext } from "react";
export const PoliceKillings2015Context = createContext();
export function loadPoliceKillings2015Data() {
  return fetch("/police-killings-2015.csv")
    .then((response) => response.text())
    .then((text) => parse(text, { columns: true }));
}

export function PoliceKillings2015Provider(props) {
  const [killings, setkillings] = useState([]);
  useEffect(() => {
    loadPoliceKillings2015Data().then((PoliceKillings2015Data) => {
      setkillings(PoliceKillings2015Data);
    });
  });
  return (
    <PoliceKillings2015Context.Provider value={[killings, setkillings]}>
      {props.children}
    </PoliceKillings2015Context.Provider>
  );
}
