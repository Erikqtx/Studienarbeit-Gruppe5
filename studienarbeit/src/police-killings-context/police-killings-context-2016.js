import parse from "csv-parse/lib/sync";
import React, { useState, useEffect, createContext } from "react";

export function loadPoliceKillings2016Data() {
  return fetch("/police-killings-2016.csv")
    .then((response) => response.text())
    .then((text) => parse(text, { columns: true }));
}

export const PoliceKillings2016Context = createContext();
export function PoliceKillings2016Provider(props) {
  const [killings, setkillings] = useState([]);
  useEffect(() => {
    loadPoliceKillings2016Data().then((PoliceKillings2016Data) => {
      setkillings(PoliceKillings2016Data);
    });
  });
  return (
    <PoliceKillings2016Context.Provider value={[killings, setkillings]}>
      {props.children}
    </PoliceKillings2016Context.Provider>
  );
}
