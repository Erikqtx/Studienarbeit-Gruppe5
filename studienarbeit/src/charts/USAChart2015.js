import React, { useContext, useMemo } from "react";
import { Bar } from "react-chartjs-2";
import { PoliceKillings2015Context } from "../police-killings-context/police-killings-context-2015";
import { extractMonths } from "../ChartHelper/FilterData";
export default function USAChart2015() {
  //load our data and save it
  const [killings] = useContext(PoliceKillings2015Context);
  //use our filter function
  const months = extractMonths(killings);
  //To get January entries
  const january = months.filter((datapoint) => datapoint === "January");
  const januaryl = january.length;
  //February
  const february = months.filter((datapoint) => datapoint === "February");
  const februaryl = february.length;
  //March
  const march = months.filter((datapoint) => datapoint === "March");
  const marchl = march.length;
  //April
  const april = months.filter((datapoint) => datapoint === "April");
  const aprill = april.length;
  //May
  const may = months.filter((datapoint) => datapoint === "May");
  const mayl = may.length;
  //June
  const june = months.filter((datapoint) => datapoint === "June");
  const junel = june.length;
  //July
  const july = months.filter((datapoint) => datapoint === "July");
  const julyl = july.length;
  //August
  const august = months.filter((datapoint) => datapoint === "August");
  const augustl = august.length;
  //September
  const september = months.filter((datapoint) => datapoint === "September");
  const septemberl = september.length;
  //October
  const october = months.filter((datapoint) => datapoint === "October");
  const octoberl = october.length;
  //November
  const november = months.filter((datapoint) => datapoint === "November");
  const novemberl = november.length;
  //December
  const december = months.filter((datapoint) => datapoint === "December");
  const decemberl = december.length;

  const data = useMemo(() => ({
    options: {
      scales: {
        yAxes: [
          {
            ticks: {
              min: 0,
            },
          },
        ],
      },
    },
    labels: [
      "January",
      "February",
      "March",
      "April",
      "May",
      "June",
      "July",
      "August",
      "September",
      "October",
      "November",
      "December",
    ],
    datasets: [
      {
        label: "Deaths per Month",
        backgroundColor: "rgba(255,99,132,0.2)",
        borderColor: "rgba(255,99,132,1)",
        borderWidth: 2,
        hoverBackgroundColor: "rgba(255,99,132,0.4)",
        hoverBorderColor: "rgba(255,99,132,1)",

        data: [
          januaryl,
          februaryl,
          marchl,
          aprill,
          mayl,
          junel,
          julyl,
          augustl,
          septemberl,
          octoberl,
          novemberl,
          decemberl,
        ],
      },
    ],
  }));

  return (
    <Bar
      data={data}
      width={700}
      height={700}
      options={{
        maintainAspectRatio: true,
        responsive: false,
        scales: {
          yAxes: [
            {
              ticks: {
                min: 70,
              },
              display: true,
            },
          ],
        },
      }}
    />
  );
}
