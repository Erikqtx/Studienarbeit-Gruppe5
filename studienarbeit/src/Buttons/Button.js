import React, { useState, render } from "react";
import "../css/Button.css";
import ReactDOM from "react-dom";
import { Button, ButtonGroup } from "reactstrap";
import PoliceKillings2015Provider from "./../police-killings-context/police-killings-context-2015";
import PoliceKillings2016Provider from "./../police-killings-context/police-killings-context-2016";
import USAChart2015 from "./../charts/USAChart2015";
import USAChart2016 from "./../charts/USAChart2016";
import Map2015 from "../Map/Map2015";
import Map2016 from "../Map/Map2016";
class LoginControl extends React.Component {
  constructor(props) {
    super(props);
    this.handleLoginClick = this.handleLoginClick.bind(this);
    this.handleLogoutClick = this.handleLogoutClick.bind(this);
    this.state = { isLoggedIn: false };
  }

  handleLoginClick() {
    this.setState({ isLoggedIn: true });
  }

  handleLogoutClick() {
    this.setState({ isLoggedIn: false });
  }

  render() {
    const isLoggedIn = this.state.isLoggedIn;
    let button;

    if (isLoggedIn) {
      button = <LogoutButton onClick={this.handleLogoutClick} />;
    } else {
      button = <LoginButton onClick={this.handleLoginClick} />;
    }

    return (
      <div>
        <Buttoncountry isLoggedIn={isLoggedIn} />
        {button}
      </div>
    );
  }
}

function Button2015(props) {
  return (
    <PoliceKillings2015Provider>
      <USAChart2015 />
      <Map2015></Map2015>
    </PoliceKillings2015Provider>
  );
}

function Button2016(props) {
  return (
    <PoliceKillings2016Provider>
      <USAChart2016 />
      <Map2016></Map2016>
    </PoliceKillings2016Provider>
  );
}

function Buttoncountry(props) {
  const isLoggedIn = props.isLoggedIn;
  if (isLoggedIn) {
    return <Button2015 />;
  }
  return <Button2016 />;
}

function LoginButton(props) {
  return <button onClick={props.onClick}>2015</button>;
}

function LogoutButton(props) {
  return <button onClick={props.onClick}>2016</button>;
}

ReactDOM.render(<LoginControl />, document.getElementById("root"));

export default LoginControl;
