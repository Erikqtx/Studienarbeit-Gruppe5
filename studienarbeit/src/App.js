import React from "react";
import "./css/App.css";
import "./Map/Map.css";
import Button from "./Button";
import MapChart from "./Map/MapChart";
import Dummy from "./dummy";
import { PoliceKillings2015Provider } from "./police-killings-context/police-killings-context-2015";
function App() {
  return (
    <div className="App">
      <div className="box header">
        <h1>
          <PoliceKillings2015Provider>
            Police Killings in America
            <hr />
            <Dummy></Dummy>
          </PoliceKillings2015Provider>
        </h1>
      </div>
      <div class="box text">
        <p>Hallo ich bin ein Text</p>
      </div>

      <div className="box map">
        <Button />
        <MapChart />
      </div>
      <div className="box info">
        <p>Info</p>
      </div>
    </div>
  );
}

export default App;
