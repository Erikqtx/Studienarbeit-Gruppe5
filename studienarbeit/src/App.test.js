import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import App from "./App";
import Button from "./Buttons/Button";
import USAChart2015 from "./charts/USAChart2015";
import { PoliceKillings2015Provider } from "./police-killings-context/police-killings-context-2015";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
});

it("matches the snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders header", () => {
  const { container } = render(<h1 />);
  container.querySelector("box header");
});

it("renders text", () => {
  const { container } = render(<p />);
  container.querySelector("box text");
});

it("renders box map", () => {
  const { container } = render(<Button />);
  container.querySelector("box map");
});

it("renders box info", () => {
  const { container } = render(
    <h2 />,
    <PoliceKillings2015Provider />,
    <USAChart2015 />
  );
  container.querySelector("box info");
});

afterEach(cleanup);
