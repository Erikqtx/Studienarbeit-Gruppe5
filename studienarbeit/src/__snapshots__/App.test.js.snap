// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render as always without data 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="box header"
    >
      <h1>
        Police Killings in America
        <hr />
        <div>
           
        </div>
        ;
      </h1>
    </div>
    <div
      class="box text"
    >
      <p>
        Hallo ich bin ein Text
      </p>
    </div>
    <div
      class="box map"
    >
      <div>
        <div
          class="button-box1"
          id="cs"
        >
          <input
            checked=""
            hidden=""
            id="country"
            name="us"
            type="radio"
          />
          <label
            for="country"
          >
            Country
          </label>
          <input
            hidden=""
            id="state"
            name="us"
            type="radio"
          />
          <label
            for="state"
          >
            State
          </label>
        </div>
        <div
          class="button-box2"
        >
          <input
            checked=""
            hidden=""
            id="year1"
            name="year"
            type="radio"
          />
          <label
            for="year1"
          >
            2015
          </label>
          <input
            hidden=""
            id="year2"
            name="year"
            type="radio"
          />
          <label
            for="year2"
          >
            2016
          </label>
        </div>
      </div>
      <svg
        class="rsm-svg "
        viewBox="0 0 800 600"
      >
        <g
          class="rsm-geographies "
        />
      </svg>
    </div>
    <div
      class="box info"
    >
      <p>
        Info
      </p>
    </div>
  </div>
</div>
`;
